pkgs/development/tools/build-managers/scala-cli/default.nix

```diff
@@ -28,7 +29,9 @@ stdenv.mkDerivation {
 
   installPhase = ''
     runHook preInstall
-    install -Dm755 scala-cli $out/bin/scala-cli
+    install -Dm755 scala-cli $out/bin/.scala-cli-wrapped
+    makeWrapper $out/bin/.scala-cli-wrapped $out/bin/scala-cli \
+      --set JAVA_HOME ${jre.home}
```

Comment:
scala-cli requires Java 17 or above, and it resolves the version based on JAVA_HOME - in order for it to see the right version I'm adding this wrapper.